google/cloud/storage/blob.py

```diff
@@ -958,6 +970,7 @@ def _do_download(
                 end=end,
                 checksum=checksum,
             )
+            download._retry_strategy = retry_strategy
```

Comment:
IIUC, this is accessing a private variable maiintained in ResumableMedia, should resumable media expose a mutable setting instead?

Reply:
Yeah, I noticed that. This is the private variable we were already manipulating in our code (that was how num_retries was implemented). I don't know the history behind that. In the interest of keeping this PR small I chose not to refactor that. If we are confident about our interface we could potentially expose this in the Download and Upload constructors instead.